src/App.tsx

```diff
@@ -5,19 +5,32 @@ import { AppStateProvider } from "./state";
 import useStateAndDispatch from "./app/hooks/useStateAndDispatch";
 import { connectWalletAndFetchBalance } from "./services/polkadotWalletServices";
 import LocalStorage from "./app/util/localStorage";
+import { createPoolCardsArray } from "./services/poolServices";
 
 const App: FC = () => {
   const { dispatch, state } = useStateAndDispatch();
-  const { api } = state;
+  const { api, pools, selectedAccount } = state;
 
   const walletConnected = LocalStorage.get("wallet-connected");
 
+  const updatePoolsCards = async () => {
+    if (api && selectedAccount && pools) {
+      await createPoolCardsArray(api, dispatch, pools, selectedAccount);
```

Comment:
What do you think about creating createPoolCardsArray with dynamic params? If you make it as dynamic, you will be able to call function one time in this case.